src/ServiceInsight/MessageList/MessageListViewModel.cs

```diff
@@ -97,76 +97,58 @@ public void CopyHeaders()
             clipboard.CopyTo(generalHeaderDisplay.HeaderContent);
         }
 
-        public void RefreshMessages(string orderBy = null, bool ascending = false)
+        public void ChangeSorting(string orderBy, bool ascending)
         {
-            var serviceControlExplorerItem = selectedExplorerItem as ServiceControlExplorerItem;
-            if (serviceControlExplorerItem != null)
+            sortColumn = orderBy;
+            sortOrderAscending = ascending;
+        }
+
+        public void NavigateToPage(string link, int pageIndex)
+        {
+            using (workNotifier.NotifyOfWork("Loading messages..."))
             {
-                RefreshMessages(searchQuery: SearchBar.SearchQuery,
-                                     endpoint: null,
-                                     orderBy: orderBy,
-                                     ascending: ascending);
+                var pagedResult = serviceControl.GetAuditMessages(link);
+
+                if (pagedResult == null)
+                {
+                    return;
+                }
+
+                pagedResult.CurrentPage = pageIndex;
+                TryRebindMessageList(pagedResult);
+
+                SearchBar.IsVisible = true;
+                SearchBar.SetupPaging(pagedResult, link);
             }
+        }
 
+        public void RefreshMessages()
```

Comment:
This is now handled in `RefreshMessages`. Navigate uses  link that SC returned in a previous search request so it does not care about initial search params.